packages/curve-ui-kit/src/features/connect-wallet/lib/ConnectionContext.tsx

```diff
@@ -120,7 +120,7 @@ export const ConnectionProvider = <
         }
 
         const walletChainId = getWalletChainId(wallet)
-        if (walletChainId && walletChainId !== chainId) {
+        if (walletChainId && walletChainId !== chainId && !document.hidden) {
```

Comment:
The `!document.hidden` looks a bit random, so a comment why this is here would be nice

Reply:
```suggestion
        const isFocused = !document.hidden // only on focused tab, so they don't fight each other
        if (walletChainId && walletChainId !== chainId && isFocused) {
```